libs/core/config/include/hpx/config/constexpr.hpp

```diff
@@ -24,12 +25,15 @@
 /// keyword. Prefer using ``constexpr`` directly instead.
 #define HPX_CONSTEXPR_OR_CONST constexpr
 
-/// This macro evaluates to ``inline constexpr`` if the compiler supports it,
-/// ``constexpr`` otherwise.
-#ifdef HPX_HAVE_CXX17_INLINE_CONSTEXPR_VARIABLE
+/// This macro evaluates to ``inline constexpr``
 #define HPX_INLINE_CONSTEXPR_VARIABLE inline constexpr
+
+/// This macro evaluates to ``inline constexpr`` for host code and
+// ``device static const`` for device code
+#if defined(HPX_COMPUTE_DEVICE_CODE)
+#define HPX_HOST_DEVICE_INLINE_CONSTEXPR_VARIABLE HPX_DEVICE static const
 #else
-#define HPX_INLINE_CONSTEXPR_VARIABLE constexpr
+#define HPX_HOST_DEVICE_INLINE_CONSTEXPR_VARIABLE HPX_INLINE_CONSTEXPR_VARIABLE
```

Comment:
I don't want to be picky here (so please ignore me if you feel I am), however I'd like to suggest to name the new variable `HPX_HOST_INLINE_CONSTEXPR_VARIABLE` as it represents an inline constexpr variable on the host only.

Reply:
`HPX_INLINE_CONSTEXPR_VARIABLE` is already `HPX_HOST_INLINE_CONSTEXPR_VARIABLE`. The point here is that it's a host and device variable so `DEVICE` really has to stay in the name. On the device it would be `HPX_DEVICE_STATIC_CONST_VARIABLE` but that's wrong on the host. So `HPX_HOST_DEVICE_VARIABLE`? That's pretty vague. `HPX_HOST_DEVICE_CONST_VARIABLE`? Not sure... Some day `inline constexpr` might actually be allowed on the device, which is why I'm not too bothered by the name.

Reply:
Fine by me, just as I said, please ignore my pickiness.